spectral_cube/spectral_cube.py

```diff
@@ -1912,6 +1912,122 @@ def subcube_from_ds9region(self, ds9region, allow_empty=False):
         # that we can now crop out
         return masked_subcube.minimal_subcube(spatial_only=True)
 
+    def subcube_from_ds9region_new(self, ds9region, allow_empty=False):
+        """
+        Extract a masked subcube from a ds9 region
+        (only functions on celestial dimensions)
+
+        Parameters
+        ----------
+        ds9region: str
+            The region to extract
+        allow_empty: bool
+            If this is False, an exception will be raised if the region
+            contains no overlap with the cube
+        """
+        import regions
+
+        if isinstance(ds9region, six.string_types):
+            ds9region = regions.DS9Parser(ds9region).shapes.to_regions()
+        else:
+            raise TypeError("{0} should be a DS9 string".format(ds9region))
+
+        return self.subcube_from_regions(ds9region, allow_empty)
+
+    def subcube_from_crtf(self, region, allow_empty=False):
+        """
+        Extract a masked subcube from a CRTF region
+        (only functions on celestial dimensions)
```

Comment:
I hope this line is no longer true when you're done

Reply:
I am not so sure, since the wcs we use for conversion to pixel region is `cube.wcs.celestial` ?

Reply:
This is what I'm talking about on the other comment: CRTFs have spectral information too, not just spatial.  For the ds9 regions, they will only function on celestial dimensions, but CRTFs will function on celestial + spectral

Reply:
oh yes ! I misread `celestial dimension` to `celestial coordinates`

Reply:
I get you! sorry I was thinking something else.